public/app/features/datasources/partials/http_settings.html

```diff
@@ -87,6 +87,19 @@ <h3 class="page-heading">Auth</h3>
 			<gf-form-checkbox class="gf-form" ng-if="current.access=='proxy'" label="Skip TLS Verify" label-class="width-10"
 			  checked="current.jsonData.tlsSkipVerify" switch-class="max-width-6"></gf-form-checkbox>
 		</div>
+		<div class="gf-form-inline">
+			<gf-form-switch class="gf-form" ng-if="current.access=='proxy'" label="Forward OAuth Identity" label-class="width-13" tooltip="Forward the user's upstream OAuth identity to the datasource (Their access token gets passed along)." label-class="width-10" checked="current.jsonData.oauthPassThru" switch-class="max-width-6"></gf-form-switch>
+		</div>
+	</div>
+
+	<div class="gf-form-group" ng-if="current.jsonData.oauthPassThru">
+		<h6>OAuth Identity Forwarding Details</h6>
```

Comment:
@bergquist Thanks for the feedback! :) Agreed, removing this option would make things more simple. However one reason I did this was because there's not a way for me to see (in the ds_proxy) what the user's currently logged-in auth_module is. 

Do you think we should add that information to `SignedInUser` or the `session.SessionStore`?

Reply:
`session.SessionStore` is about to get nuked without an imminent replacement. 

Adding it to `SignedInUser` would be nice but I think that would require an extra DB query for each request regardless if the datasource is supposed to pass those headers. I would prefer to query for the auth_module in the data proxy only if this setting is enabled. I guess we can find more performant solutions later.

Reply:
Actually I'm not sure there's a way we can tell what auth_module the user is _currently_ logged in with since that information isn't persisted anywhere. 😬 

Knowing the auth_module is required to handle token expiry/refresh, i.e `connect` in the below snippet from ds_proxy.go
```go
// TokenSource handles refreshing the token if it has expired
token, err := connect.TokenSource(proxy.ctx.Req.Context(), &oauth2.Token{
	AccessToken:  cmd.Result.OAuthAccessToken,
	Expiry:       cmd.Result.OAuthExpiry,
	RefreshToken: cmd.Result.OAuthRefreshToken,
	TokenType:    cmd.Result.OAuthTokenType,
}).Token()
```

I think the easiest solution might be to just add a `lastLogin` time to the user_auth table. Then, when we see that this datasource is using oauth_pass_thru, we can lookup the most-recent access token/auth module- which avoids us needing to have a setting for it.

What do you think about that approach? I can give it a shot and see how it feels.

Reply:
Ah, the `Created` time on that table is bumped every time the row is touched, so its basically already what I want. I'll try that first

Reply:
yea I think that's a good enough solution for now. We might want to put this into the auth token table eventually.